### PR TITLE
🔧 Add `CREATE` Tests

### PR DESCRIPTION
**Pull request by pcaversaccio**
### 🕓 Changelog

This PR adds the tests for all `CREATE` functions in `CreateX`; namely:
- `function deployCreate(bytes) payable returns (address)`
- `function deployCreateAndInit(bytes,bytes,tuple(uint256,uint256)) payable returns (address)`
- `function deployCreateAndInit(bytes,bytes,tuple(uint256,uint256),address) payable returns (address)`
- `function deployCreateClone(address,bytes) payable returns (address)`
- `function computeCreateAddress(uint256) view returns (address)`
- `function computeCreateAddress(address,uint256) view returns (address)`

I also slightly refactored the `test` directory by moving the corresponding contracts to appropriate sub-directories. Please note that this PR is rebased on the PR https://github.com/pcaversaccio/createx/pull/13. Finally, I also removed the now obsolete `CreateX.t.sol` test contract.

Eventually, I created a `CreateX` logo and added to the `README`:

#### PR Checklist

- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [x] Because this PR includes a **new feature**, the change was previously discussed in an [issue](https://github.com/pcaversaccio/createx/issues) or in the [discussions](https://github.com/pcaversaccio/createx/discussions) section. (See https://github.com/pcaversaccio/createx/issues/10.)
- [ ] I didn't do anything of this.

#### 🐶 Cute Animal Picture

image

### CODE REVIEW

File: test/mocks/ImplementationContract.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+contract ImplementationContract {
+    bool public isInitialised;
+
+    constructor() payable {}
+
+    /**
+     * @dev An initialisation function that is called once during deployment.
+     */
+    function initialiser() external payable {
+        assert(!isInitialised);
```
Comment (mds1) [inline test/mocks/ImplementationContract.sol:18]: Custom error works too if you prefer, just makes it easier to debug the line triggering a revert if we have some unique string/error
```suggestion
        require(!isInitialised, "ImplementationContract: already initialized");
```

File: test/public/CreateX.computeCreateAddress_2Args.t.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    function testFuzz_WhenTheNonceValueDoesNotExceed18446744073709551614(
+        address deployer,
+        uint64 nonce
+    ) external whenTheNonceValueDoesNotExceed18446744073709551614(deployer, nonce) {
+        vm.startPrank(deployer);
+        address createAddressComputedOnChain = address(new CreateX());
+        // It returns the 20-byte address where a contract will be stored.
+        assertEq(createX.computeCreateAddress(deployer, nonce), createAddressComputedOnChain);
+        vm.stopPrank();
```
Comment (mds1) [inline test/public/CreateX.computeCreateAddress_2Args.t.sol:34]: A nit, but might want to stop prank before the `createX.computeCreateAddress` call, to sanity check that we're not using `msg.sender` in that method

File: test/public/CreateX.deployCreate.tree [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CreateX_DeployCreate_Public_Test
+├── When the initCode creates a valid runtime bytecode
+│   ├── When the created runtime bytecode has a non-zero length
+│   │   ├── It returns a contract address with a non-zero bytecode length and a potential non-zero ether balance.
+│   │   └── It emits the event ContractCreation with the contract address as indexed argument.
+│   └── When the created runtime bytecode has a zero length
+│       └── It should revert.
+└── When the contract creation fails
+    └── It should revert.
```
Comment (mds1) [inline test/public/CreateX.deployCreate.tree:9]: wdyt of flattening this a bit to remove the very top `When`, resulting in:
- When the initCode creates runtime bytecode with a non-zero length
- When the initCode creates runtime bytecode with zero length
- When the initCode fails to deploy runtime bytecode

File: test/public/CreateX.deployCreateAndInit_3Args.tree [hunk 4]
```diff
@@ -0,0 +1,17 @@
+CreateX_DeployCreateAndInit_3Args_Public_Test
+├── When the initCode creates a valid runtime bytecode
```
Comment (mds1) [inline test/public/CreateX.deployCreateAndInit_3Args.tree:2]: Similar flattening question

File: test/public/CreateX.deployCreateAndInit_4Args.tree [hunk 5]
```diff
@@ -0,0 +1,17 @@
+CreateX_DeployCreateAndInit_4Args_Public_Test
+├── When the initCode creates a valid runtime bytecode
```
Comment (mds1) [inline test/public/CreateX.deployCreateAndInit_4Args.tree:2]: Same flattening question

File: test/mocks/ImplementationContract.sol (hunk 1 above)
Comment (pcaversaccio) [reply test/mocks/ImplementationContract.sol]: Fair, fixed in 6a79108e628e9539f349410058fa710ea570f640.

File: test/public/CreateX.computeCreateAddress_2Args.t.sol (hunk 2 above)
Comment (pcaversaccio) [reply test/public/CreateX.computeCreateAddress_2Args.t.sol]: Yeah that's a good point - I made all the `stopPrank` statements more specific now so we sanity check this everywhere: https://github.com/pcaversaccio/createx/pull/14/commits/caad20013dd604d041c8d039e262ef8734b165b3.

File: test/public/CreateX.deployCreate.tree (hunk 3 above)
Comment (pcaversaccio) [reply test/public/CreateX.deployCreate.tree]: Yeah, I think that makes sense. Implemented via ca8429d09ec5a241d42d932e5be61bfcf8cbadd9.

File: test/public/CreateX.deployCreateAndInit_3Args.tree (hunk 4 above)
Comment (pcaversaccio) [reply test/public/CreateX.deployCreateAndInit_3Args.tree]: Done via ca8429d09ec5a241d42d932e5be61bfcf8cbadd9.

File: test/public/CreateX.deployCreateAndInit_4Args.tree (hunk 5 above)
Comment (pcaversaccio) [reply test/public/CreateX.deployCreateAndInit_4Args.tree]: Done via ca8429d09ec5a241d42d932e5be61bfcf8cbadd9.

**Pull request merged**